File: src/net/eduard/api/lib/bungee/BungeeStatusUpdater.java
package net.eduard.api.lib.bungee;

import net.md_5.bungee.BungeeCord;
import net.md_5.bungee.api.config.ServerInfo;

import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.io.ObjectOutputStream;

public class BungeeStatusUpdater implements Runnable {

    public static void updateServers() {
        for (ServerSpigot server : BungeeAPI.getServers().values()) {
            ServerInfo servidor = BungeeCord.getInstance().getServerInfo(server.getName());
            if (servidor == null) continue;
            if ((!server.isDisabled())
                    && ((server.isOffline()) || ((servidor.getPlayers().size() == 0)
                    && (server.isRestarting())))) {
                servidor.ping((result, error) -> {

                    if (result == null) {
                        if (!server.isRestarting()) {
                            server.setState(ServerState.OFFLINE);
                        }
                    } else {
                        server.setMax(result.getPlayers().getMax());
                        server.setCount(result.getPlayers().getOnline());
                        if ((server.isOffline()) || (server.isRestarting())) {
                            server.setState(ServerState.ONLINE);
                        }
                    }
                });
            }
            if (!servidor.getPlayers().isEmpty()) {
                try {
                    ByteArrayOutputStream arrayOut = new ByteArrayOutputStream();
                    ObjectOutputStream out = new ObjectOutputStream(arrayOut);
                    out.writeUTF(server.getName());
                    out.writeUTF("server-update");
                    out.writeObject(server);
                    for (ServerInfo sv : BungeeCord.getInstance().getServers().values()) {
                        sv.sendData(BungeeAPI.getChannel(), arrayOut.toByteArray());
                    }
                } catch (IOException e) {
                    e.printStackTrace();
                }

            }
        }
    }

    @Override
    public void run() {
        updateServers();
    }
}

File: src/main/java/net/eduard/api/lib/bungee/BungeeStatusUpdater.kt
package net.eduard.api.lib.bungee

import java.lang.Runnable
import net.md_5.bungee.api.ProxyServer
import net.md_5.bungee.api.chat.TextComponent
import java.io.ByteArrayOutputStream
import java.io.ObjectOutputStream
import java.io.IOException

class BungeeStatusUpdater : Runnable {
    override fun run() {
        updateServers()
    }

    companion object {
        fun updateServers() {
            for (server in BungeeAPI.servers.values) {
                val servidor = ProxyServer.getInstance().getServerInfo(server.name) ?: continue
                if (server.isDisabled) continue
                servidor.ping { result, _ ->
                    if (result == null) {
                        if (!server.isRestarting) {
                            server.state = ServerState.OFFLINE
                        }
                    } else {
                        if (server.max == 1) {
                            server.max = result.players.max

                        }
                        if (!server.isGameStarted) {
                            server.state = ServerState.ONLINE
                        }
                    }
                }
                server.count = servidor.players.size
                server.players = servidor.players
                    .map { it.name }
                if (!server.changed) continue
                server.changed = false
                try {
                    ProxyServer.getInstance().console.sendMessage(
                        TextComponent("§aEnviando updates do Servidor " + server.name + " para todos servidores")
                    )
                    val arrayOut = ByteArrayOutputStream()
                    val out = ObjectOutputStream(arrayOut)
                    out.writeUTF(server.name)
                    out.writeUTF("server-update")
                    out.writeObject(server)
                    for (serverInfo in ProxyServer.getInstance().servers.values) {
                        serverInfo.sendData(
                            BungeeAPI.channel,
                            arrayOut.toByteArray()
                        )
                    }
                } catch (e: IOException) {
                    e.printStackTrace()
                }

            }
        }
    }
}